### PR TITLE
Debug plugin - fix property of non-object  when modal

### PR DESCRIPTION
**Pull request by alikon**
How to reproduce
to replicate the issue you need to have enabled log SQL executed query
see #8940 for more info
or test when modal opens like "versions"

### CODE REVIEW

**Comment by photodude** [comment]
Looks like Travis-ci checks need to rerun since several options failed the "composer install" so no tests were run.

**Comment by wilsonge** [comment]
I've given the 3 that failed a kick. But as no plugins are unit tested and the code style runs on PHP 5.6 only now (which already passes). If it fails again just ignore it and assume travis passed

**Comment by photodude** [comment]
Thanks @wilsonge looks good now.

File: plugins/system/debug/debug.php [hunk 1]
```diff
@@ -1790,21 +1790,10 @@ protected function writeToFile()
...
-		else
-		{
-			$input = $app->input;
-			$file  = $input->get('option') . $input->get('view') . $input->get('layout') . '.sql';
-			$file  = $app->get('log_path') . '/' . $domain . '_' . $file;
-		}
+		$domain = ($app->isSite()) ? 'site' : 'admin';
+		$input = $app->input;
+		$file  = $input->get('option') . $input->get('view') . $input->get('layout') . '.sql';
+		$file  = $app->get('log_path') . '/' . $domain . '_' . $file;
```
Comment (wilsonge) [inline plugins/system/debug/debug.php]: You have two variables here named file now?

**Comment by infograf768** [comment]
Works fine here.
`$conf   = JFactory::getConfig();` is not needed anymore I guess.
Also check alignment of `=`

**Comment by alikon** [comment]
ah my bad :innocent:  
code clean done

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on deee43618f5be9a6798525102073010dcd3525c3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8949">issues.joomla.org/joomla-cms/8949</a>.</sub>

**Comment by photodude** [comment]
I have tested this item :white_check_mark: successfully on deee43618f5be9a6798525102073010dcd3525c3<br /><br />:+1: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8949">issues.joomla.org/joomla-cms/8949</a>.</sub>

**Comment by infograf768** [comment]
RTC for 3.5.0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8949">issues.joomla.org/joomla-cms/8949</a>.</sub>

**Pull request merged**